pkg/ip/ip.go

```diff
@@ -804,5 +804,69 @@ func IPToPrefix(ip net.IP) *net.IPNet {
 
 // IsIPv4 returns true if the given IP is an IPv4
 func IsIPv4(ip net.IP) bool {
-	return ip.To4() != nil
+	return ip != nil && ip.To4() != nil
```

Comment:
small nit `ip != nil` is not necessary since `ip.To4()` is checking for that as well